File: Assets/CAssets/Scripts/FishingScripts/RandomMovement.js
var vx;
var vy;
var vz;
var move = true;

function Awake() {
    vx = Random.Range(-0.05, 0.05);
    vy = Random.Range(-0.01, 0.01);
    vz = Random.Range(-0.03, 0.03);
}
function Update() {
    if (move) {
        transform.position.x += vx;
        transform.position.y += vy;
        transform.position.z += vz;
    }
    else {
        transform.position.x -= vx;
        transform.position.y -= vy;
        transform.position.z -= vz;
    }

    if (transform.position.x < -5.0) {
        vx = vx * -1;
    } else if (transform.position.x > 5.0) {
        vx = vx * -1;
    }

    if (transform.position.y < -1.0) {
        vy = vy * -1;
    } else if (transform.position.y > 0.01) {
        vy = vy * -1;
    }
    if (transform.position.z < -2.0) {
        vz = vz * -1;
    } else if (transform.position.z > 2.0) {
        vz = vz * -1;
    }
    var fwd = transform.TransformDirection(Vector3.forward);
    if (Physics.Raycast(transform.position, fwd, 10)) {
        print("There is something in front of the object!");
        move = false;
    }
}

File: Assets/CAssets/Scripts/FishingScripts/RandomMovement.cs
using UnityEngine;
using System.Collections;

public class RandomMovement : MonoBehaviour {

    Rigidbody rb;
    Transform tf;
    float startY;
    float newY;

    void Start()
    {
        rb = GetComponent<Rigidbody>();
        tf = GetComponent<Transform>();
        //rb.AddRelativeTorque(Vector3.up * 1000);
        startY = rb.position.y;
    }


    void FixedUpdate()
    {
        if (rb.position.x > 5)
        {
            //startY = rb.rotation.y;
            Quaternion deltaRotation = Quaternion.Euler(new Vector3(rb.rotation.x, startY-180, rb.rotation.z) * Time.deltaTime);
            Debug.Log("fish rotation y: " + tf.rotation.y);
            //newY = Mathf.Lerp(startY, startY - 180, 5);
            if (tf.rotation.y < 278)//Mathf.FloorToInt(startY + 180)
                rb.MoveRotation(rb.rotation * deltaRotation);
            if(Mathf.FloorToInt(startY) == Mathf.FloorToInt(startY + 180))
                rb.AddRelativeForce(new Vector3(-1, 0, 0), ForceMode.Impulse);
        }
            
        
        else if(transform.position.x < -5.0)
            rb.AddRelativeForce(new Vector3(-1, 0, 0), ForceMode.Impulse);
        else
            rb.AddRelativeForce(Vector3.forward * 1, ForceMode.Impulse);
    }

    // Update is called once per frame
    void Update () {
	
	}
}
